File: summertrip/summertrip.py
idx1 = 0
idx2 = 0
count = 0
set1 = set()
string = input()
N = len(string)

while idx1 < N:
    idx2 = string.find(string[idx1], idx1 + 1)
    if idx2 != -1:  
        set1 = set(string[idx1 + 1: idx2])  
        count += len(set1)
        idx1 += 1
    else:
        set1 = set(string[idx1 + 1:])
        count += len(set1)
        idx1 += 1

print(count)    

File: summertrip/summertrip.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
// does not work yet

int main() {
    int idx1, idx2, count;
    unordered_set<char> s;  
    string input;
    cin >> input;
    int n = input.size();
    while (idx1 < n) {
        idx2 = input.find_first_of(input[idx1], idx1 + 1);
        if (idx2 != -1) {
            s = unordered_set<char>(input.begin() + idx1, input.begin() + idx2);
            count += s.size();
            idx1++;
        } else {
            s = unordered_set<char>(input.begin() + idx1, input.end());
            count += s.size();
            idx1++;
        }
    }
    cout << count << endl;
    return 0;
};
